chia/types/weight_proof.py

```diff
@@ -100,3 +103,63 @@ class WeightProof(Streamable):
     sub_epochs: List[SubEpochData]
     sub_epoch_segments: List[SubEpochChallengeSegment]  # sampled sub epoch
     recent_chain_data: List[HeaderBlock]
+
+
+@streamable
+@dataclass(frozen=True)
+class SubSlotDataV2(Streamable):
+    proof_of_space: Optional[ProofOfSpace]  # if challenge block
+    # VDF to signage point
+    cc_signage_point: Optional[VDFProof]  # if block
+    # VDF from signage to infusion point
+    cc_infusion_point: Optional[VDFProof]  # if block
+    signage_point_index: Optional[uint8]  # if block
+    cc_slot_end: Optional[VDFProof]  # if is end of slot
+    cc_sp_vdf_output: Optional[B]
+    cc_ip_vdf_output: Optional[B]
+    cc_slot_end_output: Optional[ClassgroupElement]  # if is end of slot
+    icc_infusion_point: Optional[VDFProof]
+    icc_ip_vdf_output: Optional[B]
+
+    icc_slot_end: Optional[VDFProof]  # if is end of slot
+    icc_slot_end_output: Optional[ClassgroupElement]  # if is end of slot
+    cc_sp_signature: Optional[G2Element]  # if challenge block (needed for ChallengeBlockInfo)
+    ip_iters: Optional[uint64]  # if block
+    total_iters: Optional[uint128]  # if block
+
+    def is_challenge(self) -> bool:
+        if self.proof_of_space is not None:
```

Comment:
This can be simplified:
`return self.proof_of_space is not None`
Also the same for `is_end_of_slot`